Documentation/error-details.md

```diff
@@ -0,0 +1,56 @@
+# Error Details
+
+gRPC servers return a single `error` in response to client requests. The error
+value contains a code and a description. In some cases, though, it may be
+necessary to add details about the particular error. gRPC includes a
```

Comment:
This makes it sound like gRPC has `status.Status` only because of the details.  All servers must return errors implemented by the status package or else clients will only see unknown status codes.

Would you mind renaming this file to `rpc-status.md` or `rpc-errors.md` and either:

1. Add something to the effect of: All service method handlers should return `nil` or status errors (or be coerced to UNKNOWN); clients will see those ~directly.  Maybe an example of how to use it (`status.Error(codes.Whatever, "description")`) as well.
2. Leave a placeholder for us to fill in later.

Either way, "Error Details" should ideally be just a sub-section of a bigger "Errors" doc.

Reply:
I think `rpc-errors.md` sounds good. I'll make the changes and update the PR, leaving room for a general discussion of errors.